packages/backend/src/modules/tvl2/indexers/CirculatingSupplyIndexer.ts

```diff
@@ -0,0 +1,91 @@
+import { assert } from '@l2beat/backend-tools'
+import { CoingeckoQueryService } from '@l2beat/shared'
+import { CirculatingSupplyEntry, UnixTime } from '@l2beat/shared-pure'
+import { isInteger } from 'lodash'
+
+import {
+  ManagedChildIndexer,
+  ManagedChildIndexerOptions,
+} from '../../../tools/uif/ManagedChildIndexer'
+import {
+  AmountRecord,
+  AmountRepository,
+} from '../repositories/AmountRepository'
+import { createAmountId } from '../utils/createAmountId'
+import { SyncOptimizer } from '../utils/SyncOptimizer'
+
+export interface ChainAmountIndexerDeps extends ManagedChildIndexerOptions {
+  coingeckoQueryService: CoingeckoQueryService
+  amountRepository: AmountRepository
+  syncOptimizer: SyncOptimizer
+  config: CirculatingSupplyEntry
+}
+
+export class CirculatingSupplyIndexer extends ManagedChildIndexer {
+  readonly indexerId: string
+  private readonly configId: string
+
+  constructor(private readonly $: ChainAmountIndexerDeps) {
+    super($)
+    this.indexerId = $.id
+    this.configId = createAmountId($.config)
+  }
+
+  async update(_from: number, _to: number): Promise<number> {
+    const from = this.$.syncOptimizer.getTimestampToSync(new UnixTime(_from))
```

Comment:
```suggestion
    const from = this.$.syncOptimizer.getTimestampToSync(new UnixTime(from))
```

Do not use the underscored args.